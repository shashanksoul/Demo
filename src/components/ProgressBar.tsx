import React from 'react';
import {useTranslation} from 'react-i18next';
import {ActivityIndicator, StyleSheet, Text, View} from 'react-native';

const ProgressBar: React.FC = () => {
  const {t} = useTranslation();
  return (
    <View testID="progress-bar" style={style.container}>
      <ActivityIndicator color="white" animating={true} size="small" />
      <Text style={style.label}>{t('home.progress-message')}</Text>
    </View>
  );
};

const style = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: '#FFFFFF80',
    padding: 10,
  },
  label: {color: 'white', paddingLeft: 10},
});
export default ProgressBar;
