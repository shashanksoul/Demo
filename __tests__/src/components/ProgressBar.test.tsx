import React from 'react';
import {render} from '@testing-library/react-native';
import ProgressBar from '../../../src/components/ProgressBar';

describe('ProgressBar component', () => {
  it('render', () => {
    const pBar = render(<ProgressBar />);
    expect(pBar).toMatchSnapshot();
  });
});
