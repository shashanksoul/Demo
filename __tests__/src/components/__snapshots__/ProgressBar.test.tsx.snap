// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component render 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF80",
      "flexDirection": "row",
      "padding": 10,
    }
  }
  testID="progress-bar"
>
  <ActivityIndicator
    animating={true}
    color="white"
    size="small"
  />
  <Text
    style={
      {
        "color": "white",
        "paddingLeft": 10,
      }
    }
  >
    home.progress-message
  </Text>
</View>
`;
